adr/ADR-22.md

```diff
@@ -0,0 +1,197 @@
+not need to learn which other tools might drop files elsewhere and should be
+able to, minimally, discover all the configuration by files or symbolic links
+within such directories.  For configuration files, on Unix, this means that
+all per-tool configuration will be within the `~/.config/nats/` directory,
+while managed data might be within the `~/.local/share/nats/` directory.
+Environment variables and choice of OS will change that.
+
+Our one deviation from the [adrg/xdg][] pattern is that on macOS, we
+additionally require that `~/.config/nats` work, but in new installs it should
+be a symbolic link to the `~/Library/Application Support/nats` directory.
+This is a little extra work, but warranted by both complying with platform
+expectations and being consistent since `.config/` does often exist on macOS
+platforms where users have installed additional Unix heritage tools.
+
+
+### Migration
+
+Much of the existing client configuration involves authentication data such as
+NKeys or combined `.creds` files.  These are more likely than most to have
+been moved by the users to secure storage and replaced with symbolic links or
+be on encrypted mounts.  As such, automatically migrating such data is fraught
+with peril.
+
+For the time being, we instead simply mandate that files be accessible by the
+new paths; it is acceptable for client libraries to detect that the new
+location does not exist on disk but the old location does exist, and put in a
+symbolic link at the new location pointing to the old location.
+
+Ideally, the users will migrate their data to clean up the symbolic links.
+
+At a later date, we _might_ choose to automatically migrate files, as long as
+a sweep of the old location confirms that all entries are files or
+directories, free of symlinks, and all on the same file-system.
+
+In the below descriptions of old locations, a phrase "unless overridden by an
+environment variable" would be needed in many places to strictly cover all
+contingencies; for clarity in the prose, we omit that to just describe the
+defaults.
+Similarly, unless specifically addressing an operating system, we'll use the
+XDG locations for examples.
+
+### The `nats` CLI
+
+The `nats` CLI uses the XDG location logic, in a simplified form, so the data
+has until now been written to `~/.config/nats/` always.  On Unix systems this
+remains valid; on other systems supporting symbolic links, we should point to
+that location.
+
+The `nats` CLI has proven the utility of the configuration model we're moving
+to.
+
+The configuration used by the `nats` CLI are the configuration contexts, which
+are generic for client tooling and not specific to that tool, so we consider
+the `context/` directory and `context.txt` file to be top-level configuration
+items.
+
+### The `nsc` CLI
+
+The default locations of files for `nsc` can be overridden with the
+environment variables: `$NSC_HOME`, `$NKEYS_PATH`, and `$NSC_CWD_ONLY`.
```

Comment:
$NSC_CWD_ONLY shouldn't be part of this - since that enables a feature that allows multiple terminal instances to have the right context in terms of the /Operator/Account model. This is an app feature, not some standard.